src/Ocelot/Cache/Middleware/OutputCacheMiddleware.cs

```diff
@@ -65,11 +67,54 @@ public class OutputCacheMiddleware : OcelotMiddleware
                 return;
             }
 
-            var response = HttpResponseMessage;
+            cached = await CreateCachedResponse(HttpResponseMessage);
 
-            _outputCache.Add(downstreamUrlKey, response, TimeSpan.FromSeconds(DownstreamRoute.ReRoute.CacheOptions.TtlSeconds), DownstreamRoute.ReRoute.CacheOptions.Region);
+            _outputCache.Add(downstreamUrlKey, cached, TimeSpan.FromSeconds(DownstreamRoute.ReRoute.CacheOptions.TtlSeconds), DownstreamRoute.ReRoute.CacheOptions.Region);
 
             _logger.LogDebug("finished response added to cache for {downstreamUrlKey}", downstreamUrlKey);
         }
+
+		internal Task<HttpResponseMessage> CreateHttpResponseMessage(CachedResponse cached)
+		{
+			return Task.Run(() =>
```

Comment:
Why are we spinning up a task here? I dont think we need the overhead

Reply:
Yes you're right, I originally put the Task in the signature to be uniform with CreateCachedResponse, then thought to go premature on it and spinned a task because it does a Base64 conversion.
Should I drop the Task completely? Or return a Task.FromResult() for uniformity?

Reply:
I would just drop the task!